LegionInvasionTimer.lua

```diff
@@ -186,43 +159,30 @@ frame:SetScript("OnEvent", function(f)
 		GameTooltip:Show()
 	end)
 	f:SetScript("OnLeave", GameTooltip_Hide)
-	local bg = f:CreateTexture(nil, "PARENT")
-	bg:SetAllPoints(f)
-	bg:SetColorTexture(0, 1, 0, 0.3)
-	f.bg = bg
-	local header = f:CreateFontString(nil, "OVERLAY", "TextStatusBarText")
-	header:SetAllPoints(f)
-	header:SetText(name)
-	f.header = header
-	f.startBar = startBar
-	f.db = legionTimerDB
+	f.bg = f:CreateTexture(nil, "PARENT")
+	f.bg:SetAllPoints(f)
+	f.bg:SetColorTexture(0, 1, 0, 0.3)
+	f.header = f:CreateFontString(nil, "OVERLAY", "TextStatusBarText")
+	f.header:SetAllPoints(f)
+	f.header:SetText(name)
 
 	if legionTimerDB.lock then
 		f:EnableMouse(false)
 		f.bg:Hide()
 		f.header:Hide()
 	end
 
-	candy.RegisterCallback(name, "LibCandyBar_Stop", function(_, bar, dontScan)
-		if not dontScan and bar == frame.bar1 and bar:Get("LegionInvasionTimer:complete") then
+	candy.RegisterCallback(name, "LibCandyBar_Stop", function(_, bar)
+		if bar == frame.bar[1] then
 			Timer(2, findTimer) -- Event over, start hunting for the next event
 		end
-		if bar == frame.bar1 then
-			frame.bar1 = nil
-		elseif bar == frame.bar2 then
-			frame.bar2 = nil
-		elseif bar == frame.bar3 then
-			frame.bar3 = nil
-		end
 	end)
 
 	findTimer()
 	f:RegisterEvent("SCENARIO_COMPLETED")
 	f:SetScript("OnEvent", function()
-		local _,_,_,_,_,_,_,_,_,scenarioType = C_Scenario.GetInfo()
-		if scenarioType == 4 then -- LE_SCENARIO_TYPE_LEGION_INVASION = 4
+		if select(10, C_Scenario.GetInfo()) == 4 then -- LE_SCENARIO_TYPE_LEGION_INVASION = 4
```

Comment:
This is less efficient and is actually a downgrade, all it does is make "cleaner" code, however, I do not code my addons for clean code, I code them for performance.